point_pattern.py

```diff
@@ -56,8 +57,14 @@ def find_largest_city(gj):
     population : int
                  The population of the largest city
     """
+    gj_features = gj['features']
     city = None
     max_population = 0
+    
+    for nextFeature in gj_features:
+        if gj_features[nextFeature]['pop_max'] > max_population:
```

Comment:
Here you likely do not need `gj_features[nextFeature]` as the `nextFeature` in the for loop above should be that value.  This is probably why you are getting the errors.